haskoin-wallet/Network/Haskoin/Wallet/Transaction.hs

```diff
@@ -223,15 +223,18 @@ addrTxPage (Entity ai _) addrI page@PageRequest{..}
 -- Helper function to get a transaction from the wallet database. The function
 -- will look across all accounts and return the first available transaction. If
 -- the transaction does not exist, this function will throw a wallet exception.
-getTx :: MonadIO m => TxHash -> SqlPersistT m (Maybe Tx)
-getTx txid = do
-    res <- select $ from $ \t -> do
-        where_ $ t ^. KeyRingTxHash ==. val txid
-        limit 1
-        return $ t ^. KeyRingTxTx
-    case res of
-        (Value tx:_) -> return $ Just tx
-        _ -> return Nothing
+getTx :: MonadIO m => TxHash -> SqlPersistT m (Maybe (Entity KeyRingTx))
+getTx txid = fmap listToMaybe $ select $ from $ \t -> do
+    where_ $ t ^. KeyRingTxHash ==. val txid
+    limit 1
```

Comment:
This query will only return the first transaction. The same transaction may exist in multiple accounts (especially when building offline transactions for example) so this may not be sufficient to delete all occurrences of a given TxID.